### PR TITLE
Fix ref for `npm install xx@version` installed modules

### PR DESCRIPTION
**Pull request by pangratz**
If a package has been installed via `npm install package@version`, then
`ref` and `hasVersionInRef()` returned wrong values:
- `ref` should be the version instead of `undefined`
- `hasVersionInRef()` should return `true` instead of `false`

### CODE REVIEW

**Comment by pangratz** [comment]
On a side note, this is needed because currently this "breaks" the version for `ember-data`, logged in the `console`. When `ember-data` is installed via `npm` and the recently released version `2.4.0` is used, then `2.4.0+9f8c40927a` is logged, where it should be `2.4.0`.

Since `hasVersionInRef()` is `false` [here](https://github.com/emberjs/data/blob/v2.4.0/lib/calculate-version.js#L21-L23), the abbreviated sha is added to the version.

File: test/test.js [hunk 1]
```diff
@@ -65,4 +65,36 @@ describe('npm-git-info', function() {
...
+  it('should detect info from NPM installed package without a version', function() {
+    // this tests the result of: npm install ember-data
+    var subject = info(fixture('npm-without-version.json'));
+
+    expect(subject).to.eql({
+      name: 'ember-data',
+      version: '2.4.0',
+      sha: '9f8c40927a5e8a7966c251d99eb26c3f1fb0606e',
+      abbreviatedSha: '9f8c40927a',
+      ref: undefined
```
Comment (pangratz) [inline test/test.js]: Should this be the resolved version too, so `2.4.0`? Unsure what `ref` should be here when `npm install package` is used...

Comment (tricknotes) [inline test/test.js]: It seems reasonable to be set it as `2.4.0`.
In fact, `npm install package` resolves a version as "latest" (it means "2.4.0" currently).

**Comment by tricknotes** [comment]
I left [a comment](https://github.com/tricknotes/npm-git-info/pull/1#discussion_r54555759) however this PR seems awesome :star2:

File: test/test.js (hunk 1 above)
Comment (pangratz) [inline test/test.js]: :+1: will update this PR so `2.4.0` is set...

**Comment by pangratz** [comment]
Updated PR to consider version for `ref` when using `latest`.

**Comment by tricknotes** [comment]
Thanks @pangratz :heart: 
I just released a new version as `1.0.1` :rocket:

**Comment by pangratz** [comment]
wohooo :rocket: :dancer: :dancers: :tada:

**Comment by pangratz** [comment]
Hmm, it looks like I started the dance too early. The `package.json` of a module installed via

``` json
{
  "dependencies": {
    "ember-data": "2.4.0"
  }
}
```

doesn't have a `_requested` property :cry:

@tricknotes do you have an idea what should be used in such a case? The `node_modules/ember-data/package.json` looks like this: https://gist.github.com/pangratz/a9c391037f1c43e36eb6

Maybe if no `_requested` is present, it can be assumed that `ref` is `package.version`?

**Pull request merged**